game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -156,9 +156,8 @@ boolean didAllThesePlayersJustGoToWarThisTurn(
   private boolean didThesePlayersJustGoToWarThisTurn(final PlayerId p1, final PlayerId p2) {
     // check all relationship changes that are p1 and p2, to make sure that oldRelation is not war,
     // and newRelation is war
-    for (final Tuple<Tuple<PlayerId, PlayerId>, Tuple<RelationshipType, RelationshipType>> t :
-        relationshipChangesThisTurn) {
-      final Tuple<PlayerId, PlayerId> players = t.getFirst();
+    for (final var tuple : relationshipChangesThisTurn) {
+      final Tuple<PlayerId, PlayerId> players = tuple.getFirst();
       if (players.getFirst().equals(p1)) {
```

Comment:
I see `players` is a new variable, is there variable shadowing happening here?

Reply:
The variable existed before as far as I can tell